specs/Spec.Blake2.fst

```diff
@@ -397,20 +397,22 @@ let blake2_update1 a prev m i s =
 val blake2_update_last:
     a:alg
   -> prev:nat
-  -> m:bytes{prev + length m <= max_limb a}
+  -> rem:nat
```

Comment:
I think it's conceptually a little annoying that the remainder cannot be recovered from `prev` and the length of the input, and instead has to be passed down from the caller, but perhaps that's the best way to make sure that all the functions are in agreement as to what the remainder is exactly.

Reply:
Agreed, but this will require the (fairly complicated) logic of "split" to be repeated within this function. We can revisit later if need be.